### PR TITLE
docs: add archive notice to webrtc star and direct

### PR DESCRIPTION
**Pull request by p-shahi**
Relates to: https://github.com/libp2p/github-mgmt/pull/80

### CODE REVIEW

**Comment by achingbrain** [comment]
Can we please only archive these things when there's an actual alternative for people to use?

Right now [@libp2p/webrtc-star](https://www.npmjs.com/package/@libp2p/webrtc-star) is the only way to do browser to browser communication - once the [@libp2p/webrtc]([@libp2p/webrtc](https://www.npmjs.com/package/@libp2p/webrtc) ) transport has the browser to browser component done, then we can archive WebRTC-Star but given it hasn't even made it on to npm yet this feels a little premature.

**Comment by p-shahi** [comment]
@achingbrain I see your point, the alternatives don't exist or aren't mature enough. However, the primary motivation to archive was to alleviate us of further maintenance work as that's sunken cost (this includes triaging new issues) and let the community know our intention.

If you feel this is a bit too soon, I don't see any problem unarchiving them for now.
However, archiving shouldn't prevent users from still using them and technically they can fork the repos to upstream patches (as onerous as that may be for some users.)

cc: @BigLep as he was also keen to archive the older solutions

**Comment by p-shahi** [comment]
@BigLep poke. we discussed this in last weeks SitRep but can you confirm here as well.

**Comment by BigLep** [comment]
I agree it stinks to pull things away from users, but it's being truthful.  It's not a priority for the js-libp2p maintainers to support it.  We shouldn't spend time triaging or supporting it.  Lets put that effort into the general WebRTC effort.

I think the message we want to get across crystal clear is "use at your own risk.  This is not maintained.  Here is what we're working on to replace it."

I think we're doing this, but if we need to message this better, let's do it.

That said, @p-shahi  it looks like npm hasn't been updated to make clear this is deprecated: https://www.npmjs.com/package/@libp2p/webrtc-star

Can we please do that as well?

**Comment by maschad** [comment]
Given that we released browser to browser in https://github.com/libp2p/js-libp2p-webrtc/releases/tag/v1.1.0 we should be good to merge this once https://www.npmjs.com/package/@libp2p/webrtc-star is deprecated on npm @p-shahi

**Review by achingbrain — CHANGES_REQUESTED** [review]
A couple of extra things to do here:

1. Update webrtc-direct/webrtc-star examples to use new webrtc transport
2. Sweep docs to update references to old webrtc transports to point to new version

File: examples/libp2p-in-the-browser/README.md
Comment (p-shahi) [inline examples/libp2p-in-the-browser/README.md:1]: Should we scrap the libp2p in the browser example and instead point to WebSockets, WebTransport, and WebRTC repos?

File: doc/CONFIGURATION.md
Comment (p-shahi) [inline doc/CONFIGURATION.md:862]: Should we replace this section with something other than webrtc-star?

Comment (maschad) [reply doc/CONFIGURATION.md:862]: I agree we could replace it with webrtc and it's [`DataChannelOpts`](https://github.com/libp2p/js-libp2p-webrtc/blob/bc3ae14b9ff2b78fcd35616bcd7bf76c45b7a560/src/stream.ts#L13-L17) or [Websockets and its Multiaddr filter](https://github.com/libp2p/js-libp2p-websockets/blob/3645c2ba72ff1fd0b2a3f385f3cefa5e182f27ff/src/index.ts#L21)

**Comment by p-shahi** [comment]
@maschad Can you take over this PR? i.e. replace the customizing transports with your suggestion

File: packages/transport-websockets/examples/index.js
Comment (achingbrain) [inline packages/transport-websockets/examples/index.js:14]: People copy/paste this config to get started with libp2p in the browser, I think we need to give them the best chance of success with the minimum amount of effort, so this example should have as complete a config as possible.

Please revert the removal of the extra transports.

File: packages/transport-websockets/examples/package.json
Comment (achingbrain) [inline packages/transport-websockets/examples/package.json:1]: Please keep the examples in the examples folder, it's much easier for people visiting the repo to find them that way, since they don't have to dig through the monorepo folders.

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:915]: To listen on available relay servers for incoming webrtc connections this should be:

```suggestion
    listen: ['/webrtc']
```

The relay transport should also be configured to search for relays.

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:35]: Why has this been removed?

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:7]: Why has this been removed?

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:75]: These links should point to monorepo packages where they have been migrated.

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:130]: These links should point to monorepo packages where they have been migrated.

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:134]: This is still true of modules like `kad-dht` so might be worth mentioning here.

File: doc/CONFIGURATION.md
Comment (achingbrain) [inline doc/CONFIGURATION.md:879]: Why is this config key necessary?  People will copy/paste this as good config so it's worth noting why we're overriding the default value here.  If the default value is wrong it should probably be changed instead.

File: doc/production/README.md
Comment (achingbrain) [inline doc/production/README.md:32]: It might be worth expanding this to comment on the differences between them and when you'd expect to use them.

File: examples/libp2p-in-the-browser/README.md
Comment (achingbrain) [reply examples/libp2p-in-the-browser/README.md:1]: I disagree - people getting started with libp2p in the browser use this example as a starting point so we should make it as simple, complete and easy to discover as possible.

I think linking to the universal connectivity app as a "next steps" thing would be helpful though, but it does have a whole bunch of extra stuff that can overwhelm new users.

**Review by achingbrain — CHANGES_REQUESTED** [review]

File: doc/CONFIGURATION.md
Comment (achingbrain) [reply doc/CONFIGURATION.md:7]: Right, looks like the header was removed, https://github.com/libp2p/js-libp2p/pull/1848 restores it

File: doc/CONFIGURATION.md
Comment (maschad) [reply doc/CONFIGURATION.md:879]: The purpose of this example was to hihglight that you can pass an optional `maxMessageSize` value as mentioned in the above paragraph.

File: doc/CONFIGURATION.md
Comment (maschad) [reply doc/CONFIGURATION.md:136]: I think the note is a bit confusing overall, since we configure our discovery service via a services module now, intuitively one would configure that separately from the transport. 

So I think it's best to exclude this note.

**Comment by maschad** [comment]
I've move the webRTC, websocket and webtransport examples to their own subdirectories within the `libp2p-in-the-browser` example subdirectory to make the more discoverable as @achingbrain suggested.

**Review by achingbrain — APPROVED** [review]

**Pull request merged**
